src/mp/sys/fnmp.vcxproj

```diff
@@ -28,6 +28,11 @@
     <ClCompile Include="rx.c" />
     <ClCompile Include="trace.c" />
     <ClCompile Include="tx.c" />
+    <ClInclude Include="..\..\..\inc\fnmpapi.h" />
```

Comment:
I think (though I'm not sure) we use root-relative paths rather than `..` in other project files in win-net-test

Reply:
Oh, actually, this directory should be added to the include path, rather than explicitly including each file here.

Reply:
Hum, interesting, I didn't realize it added the headers one by one.
The intended change was mostly to add a filter so that the relevant headers are easy to find. I should not have used the UX.

Reply:
Oh, ok. Toying around a bit...
- VS really wants the headers to be in the project to be able to add them in a filter (which is convenient to find the headers related to the project in the VS solution explorer)
- The full include folder is included in the "additional header" part, while the header there are the one meant to be "part of the project" (and not just part of a dependency), so it doesn't shock me to enumerate them

I'll move to absolute paths though.

Reply:
I feel pretty strongly that we should not add all of this VS goop to our source control - only the files necessary to get the code building, and ideally, opening in VS. This becomes a really big point of maintenance because the rest of our team generally does not use VS, and nothing protects these files in automation.

Additionally, it adds to the total debt checked in - why would we need to list each header included in a project, and then each header's header, etc.? AFAIK, no other build system requires this.

Reply:
Ok - if that is the general approach, I'll revert that bit, I don't see it as more than a minor convenience.

I have been using Visual Studio because the only build system provided is a Visual Studio Solution (no cmake or other tooling), so VSCode is helpless providing any IntelliSense + the `build.ps1` script takes 30 seconds for an incremental build (31.47s exactly on a re-run with 0 changes) when building the project in Visual Studio is virtually instant with changes.

I'm happy to learn if you have a trick to improve on that.

Reply:
- 1. It does work, which is why I was using it, and why I don't feel strongly about that change. I only made it because I found it more convenient to have the file listed in the UX vs having to dig for them in the list of "additional dependencies" that contains a lot of files
- 2. and 3. and 4. I'll have to look at these parameters :)
- 5. To be fair, I haven't spent much time trying to fix VSCode IntelliSense: unless I expect to work a long time on the project, if it doesn't work mostly out of the box, I don't spend the time to try to fix it
- 6. In theory, CMake has decent support (plugins understand CMake enough to get IntelliSense going, and Cmake also generate a VS solution anyway). But it might be trading one evil for another one.